### ISSUE TITLE
egl_rpi - BCMDisplayException: Failed to get display size

### ISSUE THREAD

**Issue by MrSherish** [kivy/kivy#5817]
* Python: 3.6
* OS: Raspbian Stretch Lite (without X)
* Kivy: 1.11
* Kivy installation method: pip install from master branch

### Description

I'm trying to run Kivy on Raspberry Pi Zero with OpenGL driver set to Full KMS. It fails with "egl_rpi - BCMDisplayException: Failed to get display size". Everything works when OpenGL driver is disabled.

### Code and Logs
(sorry for doubled logs)

```
[INFO   ] [Logger      ] Record log in /home/pi/.kivy/logs/kivy_18-06-22_4.txt
2018-06-22 00:10:33,265 [INFO] [kivy                    ] [Logger      ] Record log in /home/pi/.kivy/logs/kivy_18-06-22_4.txt
[INFO   ] [Image       ] Providers: img_tex, img_dds, img_sdl2, img_pil, img_gif (img_ffpyplayer ignored)
2018-06-22 00:10:33,249 [INFO] [kivy                    ] [Image       ] Providers: img_tex, img_dds, img_sdl2, img_pil, img_gif (img_ffpyplayer ignored)
[INFO   ] [Kivy        ] v1.11.0.dev0, git-4d3ea59, 20180621
2018-06-22 00:10:33,931 [INFO] [kivy                    ] [Kivy        ] v1.11.0.dev0, git-4d3ea59, 20180621
[INFO   ] [Python      ] v3.6.0 (default, Jun  4 2018, 04:51:36) 
[GCC 6.3.0 20170516]
2018-06-22 00:10:33,943 [INFO] [kivy                    ] [Python      ] v3.6.0 (default, Jun  4 2018, 04:51:36) 
[GCC 6.3.0 20170516]
[INFO   ] [Text        ] Provider: sdl2
2018-06-22 00:10:34,023 [INFO] [kivy                    ] [Text        ] Provider: sdl2
[INFO   ] [Factory     ] 194 symbols loaded
2018-06-22 00:10:34,108 [INFO] [kivy                    ] [Factory     ] 194 symbols loaded
[INFO   ] [Window      ] Provider: egl_rpi
2018-06-22 00:10:42,661 [INFO] [kivy                    ] [Window      ] Provider: egl_rpi
[INFO   ] [Window      ] Provider: sdl2(['window_egl_rpi'] ignored)
2018-06-22 00:10:43,062 [INFO] [kivy                    ] [Window      ] Provider: sdl2(['window_egl_rpi'] ignored)
[CRITICAL] [Window      ] Unable to find any valuable Window provider.
egl_rpi - BCMDisplayException: Failed to get display size
  File "/usr/local/lib/python3.6/site-packages/kivy/core/__init__.py", line 67, in core_select_lib
    cls = cls()
  File "/usr/local/lib/python3.6/site-packages/kivy/core/window/__init__.py", line 968, in __init__
    self.create_window()
  File "/usr/local/lib/python3.6/site-packages/kivy/core/window/window_egl_rpi.py", line 33, in create_window
    w, h = bcm.graphics_get_display_size(self._rpi_dispmanx_id)
  File "kivy/lib/vidcore_lite/bcm.pyx", line 74, in kivy.lib.vidcore_lite.bcm.graphics_get_display_size (/tmp/pip-a55waee4-build/kivy/lib/vidcore_lite/bcm.c:1877)

sdl2 - RuntimeError: b'No available video device'
  File "/usr/local/lib/python3.6/site-packages/kivy/core/__init__.py", line 67, in core_select_lib
    cls = cls()
  File "/usr/local/lib/python3.6/site-packages/kivy/core/window/window_sdl2.py", line 140, in __init__
    super(WindowSDL, self).__init__()
  File "/usr/local/lib/python3.6/site-packages/kivy/core/window/__init__.py", line 968, in __init__
    self.create_window()
  File "/usr/local/lib/python3.6/site-packages/kivy/core/window/window_sdl2.py", line 272, in create_window
    self.fullscreen, resizable, state)
  File "kivy/core/window/_window_sdl2.pyx", line 93, in kivy.core.window._window_sdl2._WindowSDL2Storage.setup_window (/tmp/pip-a55waee4-build/kivy/core/window/_window_sdl2.c:2398)
  File "kivy/core/window/_window_sdl2.pyx", line 66, in kivy.core.window._window_sdl2._WindowSDL2Storage.die (/tmp/pip-a55waee4-build/kivy/core/window/_window_sdl2.c:1975)

x11 - ModuleNotFoundError: No module named 'kivy.core.window.window_x11'
  File "/usr/local/lib/python3.6/site-packages/kivy/core/__init__.py", line 59, in core_select_lib
    fromlist=[modulename], level=0)

2018-06-22 00:10:43,150 [CRITICAL] [kivy                    ] [Window      ] Unable to find any valuable Window provider.
egl_rpi - BCMDisplayException: Failed to get display size
  File "/usr/local/lib/python3.6/site-packages/kivy/core/__init__.py", line 67, in core_select_lib
    cls = cls()
  File "/usr/local/lib/python3.6/site-packages/kivy/core/window/__init__.py", line 968, in __init__
    self.create_window()
  File "/usr/local/lib/python3.6/site-packages/kivy/core/window/window_egl_rpi.py", line 33, in create_window
    w, h = bcm.graphics_get_display_size(self._rpi_dispmanx_id)
  File "kivy/lib/vidcore_lite/bcm.pyx", line 74, in kivy.lib.vidcore_lite.bcm.graphics_get_display_size (/tmp/pip-a55waee4-build/kivy/lib/vidcore_lite/bcm.c:1877)

sdl2 - RuntimeError: b'No available video device'
  File "/usr/local/lib/python3.6/site-packages/kivy/core/__init__.py", line 67, in core_select_lib
    cls = cls()
  File "/usr/local/lib/python3.6/site-packages/kivy/core/window/window_sdl2.py", line 140, in __init__
    super(WindowSDL, self).__init__()
  File "/usr/local/lib/python3.6/site-packages/kivy/core/window/__init__.py", line 968, in __init__
    self.create_window()
  File "/usr/local/lib/python3.6/site-packages/kivy/core/window/window_sdl2.py", line 272, in create_window
    self.fullscreen, resizable, state)
  File "kivy/core/window/_window_sdl2.pyx", line 93, in kivy.core.window._window_sdl2._WindowSDL2Storage.setup_window (/tmp/pip-a55waee4-build/kivy/core/window/_window_sdl2.c:2398)
  File "kivy/core/window/_window_sdl2.pyx", line 66, in kivy.core.window._window_sdl2._WindowSDL2Storage.die (/tmp/pip-a55waee4-build/kivy/core/window/_window_sdl2.c:1975)

x11 - ModuleNotFoundError: No module named 'kivy.core.window.window_x11'
  File "/usr/local/lib/python3.6/site-packages/kivy/core/__init__.py", line 59, in core_select_lib
    fromlist=[modulename], level=0)

[CRITICAL] [App         ] Unable to get a Window, abort.
2018-06-22 00:10:43,173 [CRITICAL] [kivy                    ] [App         ] Unable to get a Window, abort.

```

**Comment by xaviergmail** [comment]
Also experiencing this, although on an Ubuntu armhf install.
I have tried changing the resolution using raspi-config to no avail.

**Comment by tito** [comment]
It's an issue for BCM directly. So any app using it will fail the same, not directly related to Kivy. But would be nice to understand.

**Comment by fedus** [comment]
I had the same problem, the issue was that I had the experimental OpenGL driver activated in raspi-config. Here is a write-up of how I solved it:

----------------
Running Raspbian Stretch with Python 3.5 and Kivy 1.10, I got the following error:

[CRITICAL] [Window ] Unable to find any valuable Window provider. egl_rpi - BCMDisplayException: Failed to get display size

I fixed this by DISABLING the RPI's (experimental!) GL driver (sudo raspi-config -> Advanced Options -> GL Driver -> choose Legacy).

Then, however, I got a Segmentation Fault error. But, as per https://github.com/kivy/kivy/issues/6007, the command "export KIVY_GL_BACKEND=gl" fixed this and my app runs!

Nonetheless, this feels more like a work-around, but my specific app runs.
